{
    "step_data": {
        "steps": [
            {
                "errors": [], 
                "name": "set props: revision", 
                "started": "2013-06-05 12:51:48.767751", 
                "started_linenumber": 8, 
                "finished_linenumber": 10, 
                "finished": "2013-06-05 12:51:48.768393", 
                "result": "success"
            }, 
            {
                "errors": [], 
                "name": "set props: who", 
                "started": "2013-06-05 12:51:48.768881", 
                "started_linenumber": 12, 
                "finished_linenumber": 14, 
                "finished": "2013-06-05 12:51:48.769473", 
                "result": "success"
            }, 
            {
                "errors": [
                    {
                        "line": "remoteFailed: [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionLost'>: Connection to the other side was lost in a non-clean fashion.", 
                        "linenumber": 30331
                    }, 
                    {
                        "line": "remoteFailed: [Failure instance: Traceback: <type 'exceptions.AttributeError'>: 'NoneType' object has no attribute 'callRemote'", 
                        "linenumber": 30341
                    }, 
                    {
                        "line": "exceptions.AttributeError: 'NoneType' object has no attribute 'callRemote'", 
                        "linenumber": 30558
                    }
                ], 
                "name": "'python mochitest/runtests.py ...' warnings", 
                "started": "2013-06-05 12:54:55.232364", 
                "started_linenumber": 16, 
                "finished_linenumber": 30568, 
                "finished": "2013-06-05 13:15:39.235146", 
                "result": "testfailed"
            }, 
            {
                "errors": [], 
                "name": "'rm -rf ...'", 
                "started": "2013-06-05 13:15:39.244545", 
                "started_linenumber": 30570, 
                "finished_linenumber": 30614, 
                "finished": "2013-06-05 13:16:04.014823", 
                "result": "success"
            }, 
            {
                "errors": [], 
                "name": "maybe rebooting slave lost", 
                "started": "2013-06-05 13:16:04.015405", 
                "started_linenumber": 30616, 
                "finished_linenumber": 30618, 
                "finished": "2013-06-05 13:16:04.963614", 
                "result": "success"
            }
        ], 
        "errors_truncated": false
    }, 
    "logurl": "file:///home/vagrant/treeherder/tests/sample_data/logs/tinderbox-exception.txt.gz"
}